services/localfile/server/localfile_test.go

```diff
@@ -1576,134 +1560,236 @@ func TestCopy(t *testing.T) {
 					Attrs: &pb.FileAttributes{
-						Filename: filepath.Join(temp, "/file"),
+						Filename: fmt.Sprintf("file:///some-folder"),
 					},
 				},
 				Bucket: "fil://not-a-path",
-				Key:    filepath.Base(f1.Name()),
+				Key:    "some-source-file",
 			},
-			wantErr: true,
 		},
 		{
 			name: "Bad key",
 			req: &pb.CopyRequest{
 				Destination: &pb.FileWrite{
 					Attrs: &pb.FileAttributes{
-						Filename: filepath.Join(temp, "/file"),
+						Filename: fmt.Sprintf("file:///some-folder"),
 					},
 				},
 				Bucket: "file://not-a-path",
-				Key:    filepath.Base(f1.Name()),
+				Key:    "some-source-file",
 			},
-			wantErr: true,
 		},
-		{
-			// Same here. We don't need to test all finalization permutations as TestWrite did that.
-			name: "Fail due to overwrite = false",
-			req: &pb.CopyRequest{
-				Destination: &pb.FileWrite{
-					Attrs: &pb.FileAttributes{
-						Filename: filepath.Join(temp, "/file"),
-						Attributes: []*pb.FileAttribute{
-							{
-								Value: &pb.FileAttribute_Uid{
-									Uid: uint32(uid),
-								},
+	} {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			// ARRANGE
+			client := pb.NewLocalFileClient(conn)
+
+			// ACT
+			_, err := client.Copy(context.Background(), tc.req)
+
+			// ASSERT
+			t.Log(err)
+			testutil.WantErr(tc.name, err, true, t)
+		})
+	}
+
+	t.Run("Copy a file", func(t *testing.T) {
+		// ARRANGE
+		temp := t.TempDir()
+		sourceFilename := filepath.Join(temp, "source-file")
+		sourceFileContent := "some unique content"
+		destinationFilename := filepath.Join(temp, "destination-file")
+		expectedDstFileContent := sourceFileContent
+
+		// create source file
+		sourceFile, err := os.Create(sourceFilename)
+		testutil.FatalOnErr("Create()", err, t)
+
+		// write content to source file
+		_, err = sourceFile.WriteString(sourceFileContent)
+		testutil.FatalOnErr("WriteString()", err, t)
+
+		err = sourceFile.Close()
+		testutil.FatalOnErr("Close()", err, t)
+
+		client := pb.NewLocalFileClient(conn)
+		req := &pb.CopyRequest{
+			Destination: &pb.FileWrite{
+				Attrs: &pb.FileAttributes{
+					Filename: destinationFilename,
+					Attributes: []*pb.FileAttribute{
+						{
+							Value: &pb.FileAttribute_Uid{
+								Uid: uint32(uid),
 							},
-							{
-								Value: &pb.FileAttribute_Gid{
-									Gid: uint32(gid),
-								},
+						},
+						{
+							Value: &pb.FileAttribute_Gid{
+								Gid: uint32(gid),
 							},
-							{
-								Value: &pb.FileAttribute_Mode{
-									Mode: 0777,
-								},
+						},
+						{
+							Value: &pb.FileAttribute_Mode{
+								Mode: 0777,
 							},
 						},
 					},
 				},
-				Bucket: fmt.Sprintf("file://%s", filepath.Dir(f1.Name())),
-				Key:    filepath.Base(f1.Name()),
 			},
-			filename:  filepath.Join(temp, "/file"),
-			wantErr:   true,
-			touchFile: true,
-		},
+			Bucket: fmt.Sprintf("file://%s", filepath.Dir(sourceFile.Name())),
+			Key:    filepath.Base(sourceFile.Name()),
+		}
 
-		{
-			name: "Copy a file",
```

Comment:
extract it to separate test, because it was too hard to keep so different cases in the same test